packages/iris-grid/src/IrisGridCellOverflowModal.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Editor } from '@deephaven/console';
+import { Modal, ModalBody, ModalHeader } from 'reactstrap';
+import * as monaco from 'monaco-editor';
+import { Button } from '@deephaven/components';
+import { vsJson, vsListOrdered } from '@deephaven/icons';
+import './IrisGridCellOverflowModal.scss';
+
+interface IrisGridCellOverflowModalProps {
+  text: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function IrisGridCellOverflowModal({
+  text,
+  isOpen,
+  onClose,
+}: IrisGridCellOverflowModalProps): JSX.Element {
+  const [height, setHeight] = useState(250);
+  const [showLineNumbers, setShowLineNumbers] = useState(false);
+  const [isFormatted, setIsFormatted] = useState(false);
+  const [canFormat] = useState(() => {
+    try {
+      JSON.parse(text);
+      return true;
+    } catch {
+      return false;
+    }
+  });
+  const editorRef = useRef<monaco.editor.IStandaloneCodeEditor>();
+
+  useEffect(() => {
+    if (editorRef.current) {
+      editorRef.current.layout();
+    }
+  }, [height]);
+
+  useEffect(() => {
+    if (editorRef.current) {
+      editorRef.current.updateOptions({
+        lineNumbers: showLineNumbers ? 'on' : 'off',
+      });
+    }
+  }, [showLineNumbers]);
+
+  function toggleLineNumbers() {
```

Comment:
As we discussed, `useCallback` isn't really needed in this file since re-rendering a button is super cheap and `useCallback` might even be less performant than avoiding a shallow button re-render. The button is re-rendered anyway b/c of state change so `useCallback` won't do anything other than add overhead